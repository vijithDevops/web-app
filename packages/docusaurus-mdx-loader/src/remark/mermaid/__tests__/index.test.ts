/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {createCompiler} from '@mdx-js/mdx';
import mermaid from '..';

describe('mermaid remark plugin', () => {
  function createTestCompiler() {
    return createCompiler({
      remarkPlugins: [mermaid],
    });
  }

  it("does nothing if there's no mermaid code block", async () => {
    const mdxCompiler = createTestCompiler();
    const result = await mdxCompiler.process(
      `# Heading 1

No Mermaid diagram :(

\`\`\`js
this is not mermaid
\`\`\`
`,
    );
    expect(result.contents).toMatchSnapshot();
  });

  it('works for basic mermaid code blocks', async () => {
    const mdxCompiler = createTestCompiler();
    const result = await mdxCompiler.process(`# Heading 1

\`\`\`mermaid
graph TD;
    A-->B;
    A-->C;
    B-->D;
    C-->D;
\`\`\``);
    expect(result.contents).toMatchSnapshot();
  });
});
