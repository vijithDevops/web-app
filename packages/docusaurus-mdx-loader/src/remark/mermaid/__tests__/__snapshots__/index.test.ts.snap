// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mermaid remark plugin does nothing if there's no mermaid code block 1`] = `
"


const layoutProps = {
  
};
const MDXLayout = "wrapper"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType="MDXLayout">
    <h1>{\`Heading 1\`}</h1>
    <p>{\`No Mermaid diagram :(\`}</p>
    <pre><code parentName="pre" {...{
        "className": "language-js"
      }}>{\`this is not mermaid
\`}</code></pre>
    </MDXLayout>;
}
;
MDXContent.isMDXComponent = true;"
`;

exports[`mermaid remark plugin works for basic mermaid code blocks 1`] = `
"


const layoutProps = {
  
};
const MDXLayout = "wrapper"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType="MDXLayout">
    <h1>{\`Heading 1\`}</h1>
    <mermaid {...{
      "value": "graph TD;/n    A-->B;/n    A-->C;/n    B-->D;/n    C-->D;"
    }}></mermaid>
    </MDXLayout>;
}
;
MDXContent.isMDXComponent = true;"
`;
